pyfstat/snr.py

```diff
@@ -0,0 +1,423 @@
+        stdev_2F = sqrt(8 + 4 * SNR^2)
+        ```
+
+        Input parameters are passed untouched to `self.compute_snr2`.
+        See corresponding docstring for a list of valid parameters.
+
+        Returns
+        -------
+        expected_2F:
+            Expected value of a non-central chi-squared distribution with
+            four degrees of freedom and non-centrality parameter given by SNR^2.
+        stdev_2F:
+            Standard deviation of a non-central chi-squared distribution with
+            four degrees of freedom and non-centrality parameter given by SNR^2.
+        """
+        snr2 = self.compute_snr2(*args, **kwargs)
+        expected_2F = snr2 + 4.0
+        stdev_2F = np.sqrt(8.0 + 4.0 * snr2)
+        return expected_2F, stdev_2F
+
+    def compute_Mmunu(self, Alpha, Delta):
+        """
+        Compute Mmunu matrix at a specific sky position using the detector states
+        (and possible noise weights) given at initialization time.
+        If no noise weigths were given, unit weights are assumed.
+
+        Parameters
+        ----------
+        Alpha: float
+            Right ascension (equatorial longitude) of the signal in radians.
+        Delta: float
+            Declination (equatorial latitude) of the signal in radians.
+
+        Returns
+        -------
+        Mmunu: lalpulsar.AntennaPatternMatrix
+            Mmunu matrix encoding the response of the given detector network
+            to a CW at the specified sky position.
+        """
+
+        sky = lal.SkyPosition()
+        sky.longitude = Alpha
+        sky.latitude = Delta
+        sky.system = lal.COORDINATESYSTEM_EQUATORIAL
+        lal.NormalizeSkyPosition(sky.longitude, sky.latitude)
+
+        Mmunu = lalpulsar.ComputeMultiAMCoeffs(
+            multiDetStates=self.detector_states,
+            multiWeights=self.noise_weights,
+            skypos=sky,
+        ).Mmunu
+
+        if self.noise_weights is None:
+            Mmunu.Sinv_Tsft = self.Sinv_Tsft
+
+        return Mmunu
+
+    def _convert_amplitude_parameters(self, h0, cosi, aPlus, aCross):
+        """
+        Internal method to check and convert the given amplitude parameters
+        into the required format.
+        """
+        h0_cosi = h0 is not None and cosi is not None
+        aPlusCross = aPlus is not None and aCross is not None
+
+        if h0_cosi == aPlusCross:
+            raise ValueError("Need either (h0, cosi) or (aPlus, aCross), but not both")
+
+        if h0_cosi:
+            aPlus = 0.5 * h0 * (1 + cosi**2)
+            aCross = h0 * cosi
+            return aPlus, aCross
+
+        return aPlus, aCross
+
+
+class DetectorStates:
+    """
+    Python interface to XLALGetMultiDetectorStates and XLALGetMultiDetectorStatesFromMultiSFTs.
+    """
+
+    def __init__(self):
+        self.ephems = lalpulsar.InitBarycenter(*get_ephemeris_files())
+
+    def multi_detector_states(self, timestamps, Tsft, detectors=None, time_offset=0):
```

Comment:
Didn't catch this on first round, but this should probably better be called `get_multi_detector_states`.